crates/store/src/state.rs

```diff
@@ -252,40 +268,25 @@ impl State {
             .await
             .map_err(ApplyBlockError::BlockApplyingBrokenBecauseOfClosedChannel)?;
 
+        // Awaiting the block saving task to complete without errors
+        block_save_task.await??;
+
         // scope to update the in-memory data
         {
             let mut inner = self.inner.write().await;
 
-            // Save the block before transaction is committed:
-            self.block_store.save_block(block_num, &block_data).await?;
-
             let _ = inform_acquire_done.send(());
```

Comment:
Could we add a bit more docs to this and the previous lines? If I understand correctly:

- Line 276: we lock the in-memory state; at this point all read requests to the store will start blocking.
- Line 278: we send a signal to commit the transaction to the database. We do this after locking the in-memory state so that there are no in-consistent reads where the state of the database has been updated but the in-memory state hasn't.

Reply:
Yes, you're absolutely correct. I've updated the comments, please take a look!